lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -109,7 +92,7 @@ internal sealed class MapTaskHost<TMapInput, TMapOutput> : ITask, IObserver<IClo
         /// <returns></returns>
         public byte[] Call(byte[] memento)
         {
-            while (Interlocked.Read(ref _shouldCloseTask) == 0)
+            while (!_taskCloseHandler.ShouldCloseTask())
```

Comment:
Are you talking about this line:
 while (!_taskCloseHandler.ShouldCloseTask())
There is no break keyword here.

There is a break below it:
                if (mapInput.ControlMessage == MapControlMessage.Stop)
                    {
                        break;
                    }
Do you mean here? It was original code.